producer-kafka/src/main/java/io/stargate/producer/kafka/KafkaCDCProducer.java

```diff
@@ -149,10 +168,12 @@ private void registerMetrics(MetricRegistry registry) {
 
   @Override
   public CompletableFuture<Void> close() {
-    return kafkaProducer.thenAccept(
-        producer -> {
-          producer.flush();
-          producer.close();
-        });
+    return kafkaProducer
+        .thenAccept(
+            producer -> {
+              producer.flush();
+              producer.close();
+            })
+        .thenAccept(v -> KAFKA_INIT_EXECUTOR.shutdown());
```

Comment:
Sorry, for being picky on this, but this thread pool will still apparently keep an idle thread after the `init()` is done, until the whole service is shut down, I guess :thinking: 

Since we always completely initialize the producer before registering it in this case, would it make sense to initialize it on the activator thread (`main`) and return a completed future from `init()`?

Reply:
As I pointed out in the previous comment:
https://github.com/stargate/stargate/pull/422#discussion_r526184287
We may decide to not block on init, so to fulfill the current API (returning the `CompletableFuture`) we should not block in the init() method. Other CDC producers may choose a different approach for initializing.